ports/directx-dxc/portfile.cmake

```
@@ -0,0 +1,46 @@
+vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
```

```suggestion
set(VCPKG_LIBRARY_LINKAGE dynamic)
set(VCPKG_CRT_LINKAGE dynamic)
```
I think that since this is a COM library, we should be good to simply set the values. This will enable consumption from all `x64-windows-*` triplets.

---

I was following the same pattern used for other DLL libraries that only support DLL CRT. Is this the new recommendation? @BillyONeal

---

I've confirmed that the ``DXCOMPILER.DLL`` relies on the Visual C/C++ Runtime DLLs (``MSVCP140.dll``) so this port does NOT support ``staticcrt`` scenarios. Doing otherwise would confuse end-users because the ``staticcrt`` triplets specifically don't rely on the Visual C/C++ Runtime DLLs.

The `vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)` is also more correct than using `set(VCPKG_LIBRARY_LINKAGE dynamic)` because it emits an informational message to the end-user when they are using ``static`` triplets:

```
-- Note: directx-dxc only supports dynamic library linkage. Building dynamic library.
```